mod_modular_vanilla/hooks/entity/tactical/actor.nut

```diff
@@ -358,7 +358,7 @@
 		}
 
 		// TODO: Extract into a separate function?
-		if (_hitInfo.DamageDirect < 1.0)
+		if (_hitInfo.DamageDirect < 1.0 && _hitInfo.DamageArmor > 0)	// Vanilla only checks for _hitInfo.DamageDirect here but DamageArmor might be set to 0 during some skills onBeforeDamageReceived event
```

Comment:
Probably should use `_hitInfo.DamageInflictedArmor` as that variable is supposed to represent the true damage that has gone to armor. Whereas `_hitInfo.DamageArmor` represents the incoming damage to armor.

Also, to mirror the condition for Hitpoints damage log, this should check if the actor is visible to player or not:
```squirrel
if (this.isPlayerControlled() || !this.isHiddenToPlayer())
```